### PR TITLE
symfony-cli: add build date

### LINKED ISSUE #311315: Symfony CLI version command shows current time as build date
### Describe the bug
After installing `symfony-cli` you can run `symfony version`. This command prints something like:
`Symfony CLI version 5.8.17 (c) 2021-2024 Fabien Potencier (2024-05-13T12:16:30+02:00 - stable)`

Usually the date at the end is the build date. But when installed via nix it always shows the current time, so the command output is different for every second.

The build date is passed here:
https://github.com/symfony-cli/symfony-cli/blob/870a3ce2ccaa6428f83428e204231b0c9be9b698/main.go#L126

It also has a comment saying `overridden at linking time`:
https://github.com/symfony-cli/symfony-cli/blob/870a3ce2ccaa6428f83428e204231b0c9be9b698/main.go#L43-L44

### Steps To Reproduce
1. Install symfony-cli: https://search.nixos.org/packages?channel=unstable&show=symfony-cli&from=0&size=50&sort=relevance&type=packages&query=symfony-cli
2. Run `symfony version`

### Expected behavior
It should show the real build time of that release (time of last commit?).

### Notify maintainers
@drupol

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.5.0-28-generic, Linux Mint, 21.3 (Virginia), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.22.0`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/home/rafael/.nix-defexpr/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by drupol**
This fix https://github.com/NixOS/nixpkgs/issues/311315

Inspired from `pkgs/applications/graphics/pdfcpu/default.nix`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by patka-123 — APPROVED** [review]

File: pkgs/by-name/sy/symfony-cli/package.nix [hunk 1]
```diff
@@ -17,7 +17,13 @@ buildGoModule rec {
     owner = "symfony-cli";
     repo = "symfony-cli";
     rev = "v${version}";
-    hash = "sha256-IanaxFhD0nAabr9w6ARCVie+sYW9bvgHoahsuHQYeqE=";
+    hash = "sha256-UmGyIZk5s5A8ModafWMZqeJHdZ4fa+hAHi62pdlfJ8I=";
+    leaveDotGit = true;
+    postFetch = ''
+      cd "$out"
+      git log -1 --pretty=%cd --date=format:'%Y-%m-%dT%H:%M:%SZ' > $out/SOURCE_DATE
+      find "$out" -name .git -print0 | xargs -0 rm -rf
```
[the remark this answers is not in the document]
Comment (drupol) [reply pkgs/by-name/sy/symfony-cli/package.nix]: Yes thanks!

File: pkgs/by-name/sy/symfony-cli/package.nix [hunk 2]
```diff
@@ -27,6 +33,10 @@ buildGoModule rec {
     "-X main.channel=stable"
   ];
 
+  preBuild = ''
+    ldflags+=" -X main.buildDate=$(cat SOURCE_DATE)"
```
[the remark this answers is not in the document]
Comment (drupol) [reply pkgs/by-name/sy/symfony-cli/package.nix:37]: No, it's not necessary. The `$src` is not copied to `$out`, it's only the result of the compilation of `$src` which is copied to `$out`.

File: pkgs/by-name/sy/symfony-cli/package.nix [hunk 3]
```diff
@@ -17,7 +17,13 @@ buildGoModule rec {
     owner = "symfony-cli";
     repo = "symfony-cli";
     rev = "v${version}";
-    hash = "sha256-IanaxFhD0nAabr9w6ARCVie+sYW9bvgHoahsuHQYeqE=";
+    hash = "sha256-UmGyIZk5s5A8ModafWMZqeJHdZ4fa+hAHi62pdlfJ8I=";
+    leaveDotGit = true;
+    postFetch = ''
+      cd "$out"
+      git log -1 --pretty=%cd --date=format:'%Y-%m-%dT%H:%M:%SZ' > $out/SOURCE_DATE
```
[the remark this answers is not in the document]
Comment (drupol) [reply pkgs/by-name/sy/symfony-cli/package.nix]: I tried it, it doesn't work and I can't explain why TBH.

[the remark this answers is not in the document]
Comment (drupol) [reply pkgs/by-name/sy/symfony-cli/package.nix]: Indeed, that works! Thanks :)

File: pkgs/by-name/sy/symfony-cli/package.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (drupol) [reply pkgs/by-name/sy/symfony-cli/package.nix:36]: It's in `$out` of `$src`, not in `$out` of the main derivation.

**Pull request merged**